src/main/323C.c

```diff
@@ -1366,9 +1504,169 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002A6FC);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002A728);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002A74C);
+void func_8002A74C()
+{
+    s8 fill = 0;
+    s32 a2;
+    s32 var_v1;
+    s8* var_a0;
+    s8* a0;
+    u32 a1;
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002A7D0);
+    for (a1 = 0; a1 < COUNT(game_objects); a1++) {
+        a0 = (u8*)&game_objects[a1];
+        var_v1 = sizeof(game_objects[0]) - 1;
+        do {
+            *a0++ = fill;
+        } while (var_v1-- != 0);
+    }
+
+    for (a1 = 0; a1 < COUNT(shot_objects); a1++) {
+        a0 = (u8*)&shot_objects[a1];
+        var_v1 = sizeof(shot_objects[0]) - 1;
+        do {
+            *a0++ = fill;
+        } while (var_v1-- != 0);
+    }
+}
+
+extern s32 D_8013E188;
+extern s32 D_8013E18C;
+extern s32 D_8013E190;
+extern s32 D_8013E194;
+extern s8 D_8013E1BC;
+extern s16 D_8013E1BE;
+extern struct UnkObj D_80141AB0[];
+extern s8 D_80141BE4;
+extern s16 D_80141BE6;
+extern s8 D_801754A0;
+extern s8 D_80175E9C;
+extern s16 D_80175EA0;
+
+// some kind of reset?
+void func_8002A7D0(void)
+{
+    s8 fill = 0;
+    s32 a2;
+    s32 var_v1;
+    s8* var_a0;
+    s8* a0;
+    u32 a1;
+
+    func_8002A728(&D_801418C8);
+    func_8002A728(&D_80175D58);
+
+    for (a1 = 0; a1 < 3; a1++) {
+        a0 = (u8*)&D_80141AB0[a1];
+        var_v1 = 0x60 - 1;
+        do {
+            *a0++ = fill;
+        } while (var_v1-- != 0);
+    }
+
+    for (a1 = 0; a1 < 2; a1++) {
+        a0 = (u8*)&D_8013E470 + a1 * 0x50;
```

Comment:
This one is probably a struct of size 0x50 or 0x28 but I need to first track down where else it's referenced.